### PR TITLE
feat: mev landing

### PR DESCRIPTION
**Pull request by Chef-Yogi**
---

## PR-Codex overview
This PR introduces a new feature for the PancakeSwap platform, focusing on implementing MEV (Miner Extractable Value) protection. It adds interfaces, components, and functionalities to safeguard users from frontrunning and sandwich attacks during swaps.

### Detailed summary
- Added `PageProps` and `RPCResponse` interfaces in `apps/web/src/views/Mev/types.ts`.
- Introduced `ShieldIcon` in `packages/uikit/src/components/Svg/Icons/ShieldIcon.tsx`.
- Created `MevLanding` component in `apps/web/src/views/Mev/index.tsx`.
- Implemented `AdMevProtection` component in `apps/web/src/components/AdPanel/Ads/AdMevProtection.tsx`.
- Integrated `MevToggle` to enable/disable MEV protection.
- Added `MevSwapDetail` component to display MEV status.
- Updated `V4SwapForm` to include `mevSlot` prop for MEV toggle.
- Created utility functions in `apps/web/src/views/Mev/utils.ts` for fetching RPC data.
- Added new wallet integration guides in `apps/web/src/views/Mev/InfoSection.tsx`.
- Enhanced UI with animations and styling for MEV-related components.
- Updated localization for new MEV-related texts in `packages/localization/src/config/translations.json`.

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by ChefMomota** [comment]
Screenshot 2024-11-20 at 9 49 33 AM
Dark theme can't see the word. Do we need set max width to 1200px?

File: apps/web/src/views/Mev/Hero.tsx [hunk 1]
```diff
@@ -0,0 +1,328 @@
...
+                <HeroWalletImg src={getImageUrl('hero-wallet.png')} alt="hero-wallet" />
+              </FlexGap>
+              <FlexGap flexDirection="column" justifyContent="space-between">
+                <Box>
+                  <Text fontSize="32px" lineHeight="38px" bold mb="8px">
+                    {t('In one click')}
+                  </Text>
+                  <Text>{t('Add automatically on BNB Smart Chain: PancakeSwap MEV Guard ')}</Text>
+                </Box>
+                <ConnectWalletButton withIcon />
```
Comment (ChefMomota) [inline apps/web/src/views/Mev/Hero.tsx:305]: const { account } = useAccountActiveChain()

{account &&  <ConnectWalletButton withIcon />}

File: apps/web/src/views/Mev/InfoSection.tsx [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+                  {t(wallet.title)}
+                </Text>
+              </FlexGap>
+            ))}
+          </FlexGap>
+          <FlexGap flexDirection="column" gap="24px" justifyContent="center" alignItems="center">
+            <Text>{t('or')}</Text>
+            <Button
+              onClick={() => {
+                window.open('https://docs.pancakeswap.finance/', '_blank', 'noopener noreferrer')
```
Comment (ChefMomota) [inline apps/web/src/views/Mev/InfoSection.tsx:109]: Do we have a doc url for this?
Or this is the actual url we need redirect?

**Comment by ChefMomota** [comment]
Do we need to add button in menu to direct to `/mev` ?

File: apps/web/src/views/Mev/MevToggle.tsx [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+        <Toggle scale="md" checked={isMEVEnabled} onClick={onOpen} />
+      </ToggleWrapper>
+      <MevModal isOpen={isOpen} onDismiss={onDismiss} />
+    </>
+  )
+}
+
+export const MevModal: React.FC<{ isOpen: boolean; onDismiss?: () => void; onSuccess?: () => void }> = ({
+  isOpen,
+  onDismiss,
```
Comment (ChefMomota) [inline apps/web/src/views/Mev/MevToggle.tsx:78]: `onDismiss` we can use this type
import { InjectedModalProps } from '@pancakeswap/uikit'

File: apps/web/src/views/Mev/MevIntroSection.tsx [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+            <Card>
+              <Text maxWidth="60%">
+                {t('Enjoy safe, secure and private Swaps without frontrunning and sandwich attacks.')}
+              </Text>
+              <ImageBox src={getImageUrl('card-1.png')} />
+            </Card>
+            <Card>
+              <Text maxWidth="60%">
+                {t('Fast and reliable RPC endpoint for your daily usage, beyond Swapping and trading.')}
+              </Text>{' '}
```
Comment (ChefMomota) [inline apps/web/src/views/Mev/MevIntroSection.tsx:119]: remove `{' '}`
change it to `<Text maxWidth="60%" mr="4px"></Text>`

**Comment by ChefMomota** [comment]
Dark mode icon color
Screenshot 2024-11-28 at 1 18 58 PM

**Comment by thechefpenguin** [comment]
image

Ideally the icons should be clickable as well. Right now only wallet name is clickable

**Review by ChefMomota — DISMISSED** [review]

**Comment by thechefpenguin** [comment]
The title says "MEV GARD" is this intentional or a typo of "Guard"

File: apps/web/src/views/Mev/MevToggle.tsx (hunk 3 above)
Comment (Chef-Yogi) [reply apps/web/src/views/Mev/MevToggle.tsx:80]: done

File: apps/web/src/views/Mev/MevIntroSection.tsx (hunk 4 above)
Comment (Chef-Yogi) [reply apps/web/src/views/Mev/MevIntroSection.tsx]: removed

**Comment by Chef-Yogi** [comment]
@thechefpenguin thanks for the reminder, fixed

**Review by ChefMomota — APPROVED** [review]

**Pull request merged**